src/framework/piral-core/src/modules/element.test.ts

```diff
@@ -0,0 +1,29 @@
+import { Atom, deref } from '@dbeining/react-atom';
+import { createListener } from 'piral-base';
+import { createActions } from '../state';
+import { renderElement } from './element';
+
+describe('Elemnts Module', () => {
+  it('renderElement', () => {
+    const state = Atom.of({
+      portals: {},
+    });
+    const context = createActions(state, createListener({}));
+    const element = document.createElement('piral-extension');
+    document.body.appendChild(element);
+    element.params;
```

Comment:
Missing assertions (*why* are you reading the properties without storing / evaluating their current values?).